community_server/src/Template/AppRequests/list_transactions.ctp

```diff
@@ -0,0 +1,17 @@
+<?php
+
+/* 
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+$body['balance'] = $this->element('centToFloat', ['cent' => $body['balance'], 'precision' => 4]);
+$body['decay'] = $this->element('centToFloat', ['cent' => $body['decay'], 'precision' => 4]);
+$body['gdtSum'] = $this->element('centToFloat', ['cent' => $body['gdtSum'], 'precision' => 2]);
+
+foreach($body['transactions'] as $i => $transaction) {
+    $body['transactions'][$i]['balance'] = $this->element('centToFloat', ['cent' => $transaction['balance'], 'precision' => 4]);
+}
```

Comment:
this seems like a hack.

Am I right that this comes directly from the database and later on is transformed with this loop?
Or is it that the `$body['transactions']`is build by us in the code?

The ideal way would be:

```
$result = []
while(i can read from db){
   $result[] = transform(db.row)
}
```
That way we would only have 1 loop. This is more about effectiveness then correctness, therefore you can neglect this. If we build the transactions array ourselves with a loop somewhere in the code, I would suggest to move the `centToFloat` call there.

Reply:
CakePHP is a MVC-Framework, you suggestet that I use the transform code only once. 
I can use Component which all Controller can use, I can use a Behaviour which all Model Entitys can use or can make a element which all views can use. The code which build together the transaction list is splittet in Controller and Model Table Classes.
So the best solution from point of view was to use the view. It's belong naturally there because it is only for the user to see. 
So I must use a additional loop in view to transform. 
Yes from performance point of view it is not optimal, from performant point of view my first approach was optimal, but you don't like it.

Reply:
It don't came directly from db, it is builded by Model/Table/Transactions::listTransactionsHumanReadable

Reply:
Actually I would change code on 4 places when I make the transformation direct in Model/Table/Transactions::listTransactionsHumanReadable in the loop.

Reply:
Plus I must make a new transformation function because I cannot call the element inside a model table class.

Reply:
Its fine, thanks for clarifying ❤️